zerver/migrations/0422_backfill_realmfilter_url_template.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 4.0.7 on 2022-10-02 20:50
+
+import re
+
+import uri_template
+from django.db import migrations
+from django.db.backends.postgresql.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def transform_to_url_template_syntax(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    linkifier_model = apps.get_model("zerver", "RealmFilter")
+
+    linkifiers = linkifier_model.objects.all()
+    # Matches anything of the form %(variable_name)
+    var_pattern = re.compile(r"%\(([a-zA-Z0-9_-]+)\)s")
+    for linkifier in linkifiers:
+        # Replace format string variables with the RFC 6570 URI Template syntax
+        converted_template = var_pattern.sub(r"{\1}", linkifier.url_format)
```

Comment:
This doesn’t correctly handle patterns containing `%%`, which is an escaped `%` in the language of format strings. We need to convert

`http://foo/%%` → `http://foo/%`
`http://foo/%%%%` → `http://foo/%%`
`http://foo/%%%%%%` → `http://foo/%%%`

`http://foo/%(var)s` → `http://foo/{var}`
`http://foo/%%(var)s` → `http://foo/%(var)s`
`http://foo/%%%(var)s` → `http://foo/%{var}`
`http://foo/%%%%(var)s` → `http://foo/%%(var)s`
`http://foo/%%%%%(var)s` → `http://foo/%%{var}`
`http://foo/%%%%%%(var)s` → `http://foo/%%%(var)s`

Reply:
We also need to quote any pre-existing `{` and `}` as `%7B` and `%7D`.